### PR TITLE
feat(icon): Add Icon component and registry

### PR DESCRIPTION
**Pull request by thomaspink**
Issue: #13

### CODE REVIEW

File: src/lib/icon/icon-registry.ts [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+}
+
+/** Configuration for an icon, including the URL and possibly the cached SVG element. */
+interface SvgIconConfig {
+  svgElement: SVGElement | null;
+  url: SafeResourceUrl;
+}
+/**
+ * Creates a new config object without setting svgElement
+ * We use a function for creating a config object instad of a class to
```
Comment (ffriedl89) [inline src/lib/icon/icon-registry.ts]: typo instad = > instead

File: src/demo-app/icon/assets/angularjs.svg [hunk 2]
```diff
@@ -0,0 +1,14 @@
+
+            <!--?xml version="1.0" encoding="utf-8"?-->
```
Comment (ffriedl89) [inline src/demo-app/icon/assets/angularjs.svg]: we should only use optimized svgs https://jakearchibald.github.io/svgomg/

File: src/lib/icon/icon.scss [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+@mixin square($side) {
+  width: $side;
+  height: $side;
+}
+
+.gh-icon {
+  display: inline-block;
+}
+
+.gh-icon-small {
```
Comment (ffriedl89) [inline src/lib/icon/icon.scss]: should we use bem syntax .gh-icon--small?

Comment (thomaspink) [reply src/lib/icon/icon.scss]: Should be discussed in its own issue

**Pull request merged**